### PR TITLE
DACT-1234: Additional Validation For Nationality

### PR DESCRIPTION
**Pull request by olaych**
Jira: https://companieshouse.atlassian.net/browse/DACT-1234
Added nationality validation
Added tests

### CODE REVIEW

File: src/validation/nationality.validation.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  if((nationality[0] && nationality[1]) && (nationality[1] ===  nationality[0])) {
+    return nationaliltyValidationType.DuplicatedNationality2Validator.Nationality
+  }
+  if((nationality[0] && nationality[1] && nationality[2]) && (nationality[2] ===  nationality[0] || nationality[2] === nationality[1])) {
+    return nationaliltyValidationType.DuplicatedNationality3Validator.Nationality
+  }
+}
+
+export const validateMaximumLengthNationality = (nationality: string[], nationaliltyValidationType: NationalityValidationType) => {
+  if ((nationality) && (`${nationality[0]},${nationality[1]},${nationality[2]}`.length > 48)) {
```
Comment (jbestwickch) [inline src/validation/nationality.validation.ts:131]: In the API we allow 49 characters for 2 nationalities.

Comment (olaych) [reply src/validation/nationality.validation.ts:131]: Dual nationality character added

File: test/controllers/director.nationality.controller.unit.ts [hunk 2]
```diff
@@ -82,14 +84,20 @@ describe("Director nationality controller tests", () => {
...
+        
         //mockGetField.mockReturnValue("dj");
         const response = await request(app).post(DIRECTOR_NATIONALITY_URL).send({typeahead_input_0:"dj",typeahead_input_1:"dj",typeahead_input_2:"dj"});
   
         expect(response.text).toContain("Select a nationality from the list");
         expect(response.text.includes("For technical reasons, we are currently unable to accept multiple nationalities with a total of more than 48 characters")).toEqual(false);
         expect(response.text).toContain("John Smith");
-        expect(mockGetValidationStatus).toHaveBeenCalled();
-        expect(mockPatchOfficerFiling).toHaveBeenCalled();
+        // expect(mockGetValidationStatus).toHaveBeenCalled();
```
Comment (anona-w) [inline test/controllers/director.nationality.controller.unit.ts:99]: Comment should probably be removed

Comment (olaych) [reply test/controllers/director.nationality.controller.unit.ts]: Comment fixed

**Review by jbestwickch — DISMISSED** [review]

**Review by clewis1 — APPROVED** [review]
Looks good.

**Review by jbestwickch — APPROVED** [review]

**Review by clewis1 — APPROVED** [review]

**Pull request merged**